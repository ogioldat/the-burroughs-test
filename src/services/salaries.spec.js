const { getSalaries } = require('./salaries')

describe('salaries service', () => {
    describe('getSalaries', () => {
        test('it returns salaries for correct date', async () => {
            const date = new Date('2021-08-22')
            const salaries = getSalaries({ date })

            expect(salaries).toMatchSnapshot(
                `"baseSalary","bonusSalary"
                "2021-08-31","2021-09-15"
                "2021-09-30","2021-10-15"
                "2021-10-29","2021-11-15"
                "2021-11-30","2021-12-15"
                "2021-12-31","2022-01-19"
                "2022-01-31","2022-02-15"
                "2022-02-28","2022-03-15"
                "2022-03-31","2022-04-15"
                "2022-04-29","2022-05-18"
                "2022-05-31","2022-06-15"
                "2022-06-30","2022-07-15"
                "2022-07-29","2022-08-15"
            `
            )
        })

        test('it validates input', async () => {
            const date = new Date('2021-13-3')

            await expect(getSalaries({ date })).rejects.toMatchObject({
                message: 'Invalid time value',
            })
        })
    })
})
