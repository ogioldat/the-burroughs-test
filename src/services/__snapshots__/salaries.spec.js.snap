// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`salaries service getSalaries it returns salaries for correct date: "baseSalary","bonusSalary"
                "2021-08-31","2021-09-15"
                "2021-09-30","2021-10-15"
                "2021-10-29","2021-11-15"
                "2021-11-30","2021-12-15"
                "2021-12-31","2022-01-19"
                "2022-01-31","2022-02-15"
                "2022-02-28","2022-03-15"
                "2022-03-31","2022-04-15"
                "2022-04-29","2022-05-18"
                "2022-05-31","2022-06-15"
                "2022-06-30","2022-07-15"
                "2022-07-29","2022-08-15"
             1`] = `
"\\"baseSalary\\",\\"bonusSalary\\"
\\"2021-08-31\\",\\"2021-09-15\\"
\\"2021-09-30\\",\\"2021-10-15\\"
\\"2021-10-29\\",\\"2021-11-15\\"
\\"2021-11-30\\",\\"2021-12-15\\"
\\"2021-12-31\\",\\"2022-01-19\\"
\\"2022-01-31\\",\\"2022-02-15\\"
\\"2022-02-28\\",\\"2022-03-15\\"
\\"2022-03-31\\",\\"2022-04-15\\"
\\"2022-04-29\\",\\"2022-05-18\\"
\\"2022-05-31\\",\\"2022-06-15\\"
\\"2022-06-30\\",\\"2022-07-15\\"
\\"2022-07-29\\",\\"2022-08-15\\""
`;
